spec/api/datastore/maat/maat_ready_application_spec.rb

```diff
@@ -0,0 +1,77 @@
+require 'rails_helper'
+
+RSpec.describe 'get maat ready application' do
+  describe 'GET /api/maat/applications/:usn' do
+    let(:api_request) { get "/api/maat/applications/#{application_usn}" }
+
+    let(:application_usn) { application.application['reference'] }
+
+    let(:application) do
+      CrimeApplication.new(
+        application: JSON.parse(LaaCrimeSchemas.fixture(1.0).read),
+        review_status: :ready_for_assessment,
+      )
+    end
+
+    context 'with a ready for assessment application' do
+      before do
+        allow(CrimeApplication).to receive(:find_by)
+          .with({ reference: application_usn })
+          .and_return(application)
+
+        api_request
+      end
+
+      it 'returns http status 200' do
+        expect(response).to have_http_status(:success)
+      end
+
+      it 'returns the application details' do
+        expect(JSON.parse(response.body)).to match(application.application)
+      end
+    end
+
+    context 'with a completed application' do
+      before do
+        application.update!(review_status: :assessment_completed, reviewed_at: 1.week.ago)
+        api_request
+      end
+
+      it_behaves_like 'an error that raises a 409 status code'
+    end
+
+    context 'with a returned application' do
+      before do
+        application.update!(review_status: :returned_to_provider, reviewed_at: 1.week.ago,
+                            returned_at: 1.week.ago)
+        api_request
+      end
+
+      it_behaves_like 'an error that raises a 409 status code'
+    end
+
+    context 'with a received application' do
+      before do
+        application.update!(review_status: :application_received)
+        api_request
+      end
+
+      it_behaves_like 'an error that raises a 409 status code'
+    end
+
+    context 'when not found' do
+      before do
+        allow(CrimeApplication).to receive(:find_by)
+          .with({ reference: application_usn }) {
```

Comment:
We can remove the hash literal here `{...}` as with will take the arguments as a hash by default:

`.with(reference: application_usn)` is functionally the same 👍